src/treetensornetworks/abstracttreetensornetwork.jl

```diff
@@ -365,20 +365,14 @@ end
 # Inner products
 #
 
-# TODO: implement using multi-graph disjoint union
-function inner(
-  y::AbstractTTN, A::AbstractTTN, x::AbstractTTN; root_vertex=default_root_vertex(x, A, y)
-)
-  traversal_order = reverse(post_order_dfs_vertices(x, root_vertex))
-  check_hascommoninds(siteinds, A, x)
-  check_hascommoninds(siteinds, A, y)
+# TODO: Remove dispatch on AbstractTTN (unless we want to trigger warning for trees if BP is nonexact)  
+function inner(y::AbstractTTN, A::AbstractTTN, x::AbstractTTN; kwargs...)
   ydag = sim(dag(y); sites=[])
-  x = sim(x; sites=[])
-  O = ydag[root_vertex] * A[root_vertex] * x[root_vertex]
-  for v in traversal_order[2:end]
-    O = O * ydag[v] * A[v] * x[v]
-  end
-  return O[]
+  blf = BilinearFormNetwork(A, ydag, x)
```

Comment:
I don't think we should be using a form here, that is mostly meant for optimization.

Reply:
Is your goal to make a tensor network out of the network `y`, `A`, and `x`? If so, you can use `disjoint_union`.

Reply:
Yes, as far as I understand the `BilinearFormNetwork` just stacks the network here.

Reply:
Ok, let's do that without going through `BilinearFormNetwork`.